### ISSUE TITLE
Nested FSM not saving

### ISSUE THREAD

**Issue by robnauticus** [imjp94/gd-YAFSM#7]
Hello, thanks for the plugin! Enjoying it so far. My root flow is working well.

I've got a simple state flow and I'm trying to add a nested state machine to one of the nodes. When I double click to edit the nested flow, everything seems to indicate that it is working properly. However when I save and leave the project, the nested nodes are gone. Completely empty as if never opened before. Not even the entry and exit nodes remain.
![image](https://user-images.githubusercontent.com/10800102/108102695-b8c27600-703d-11eb-93e2-7816303df4c5.png)
![image](https://user-images.githubusercontent.com/10800102/108102949-1b1b7680-703e-11eb-95c7-3c93de9bc471.png)

I'm simply double clicking the CombatAction_Npc node in the middle. Adding entry and exit points with a single transition. I also tried a more complex one at first but when I came back and it was gone, I decided to pull back and let you know.

Thanks!

**Comment by imjp94** [comment]
1) May I know how the resource are saved? [Built-in or external](https://docs.godotengine.org/en/stable/getting_started/step_by_step/resources.html#external-vs-built-in)?
Built-in resource will be saved automatically as you save the scene.
In case of saving external resource, a dialog will popup:
![Screenshot 2021-02-17 023250](https://user-images.githubusercontent.com/11486079/108106086-74c88300-70c8-11eb-980e-d16caaf21dc7.png)

2) Does the parent node turns into hard-corner rectangle(which indicate a nested node)?:
![Screenshot 2021-02-17 022343](https://user-images.githubusercontent.com/11486079/108105182-41d1bf80-70c7-11eb-8911-a555fc0848d3.png)

**Comment by robnauticus** [comment]
Looks like I got it. It was an instance issue with the resource in my scene. I had to make the changes to the StateMachinePlayer.tscn scene then re-add it to my actual scene. Seems to be saving and linked up properly once I selected the saved resource I called StateMachinePlayer.tres. Thanks for the help!

Yes it does properly show up now as a nested node. I'm even trying to go another level after that.
